catalog/core/catalog-core-api-impl/src/test/java/ddf/catalog/data/impl/MetacardImplTest.java

```diff
@@ -562,4 +566,39 @@ public void testDeserializingHiddenAttribute() throws IOException, ClassNotFound
 
     }
 
+    @Test
+    public void testSourceIdSerialization() throws Exception {
+        String sourceId = "TheSourceIsStrongWithThisOne";
+
+        MetacardImpl metacard = new MetacardImpl();
+        metacard.setSourceId(sourceId);
+
+        byte[] serializedMetacard;
+        try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                ObjectOutputStream oos = new ObjectOutputStream(baos);) {
+            oos.writeObject(metacard);
+            serializedMetacard = baos.toByteArray();
+        }
+
+        Metacard deserializedMetacard;
+        try (ByteArrayInputStream bais = new ByteArrayInputStream(serializedMetacard);
+                ObjectInputStream ois = new ObjectInputStream(bais);) {
+            deserializedMetacard = (Metacard) ois.readObject();
+        }
+
+        assertEquals(deserializedMetacard.getSourceId(), sourceId);
```

Comment:
Currently the rest file is using the old assertEquals, so keeping it in the same format as the other tests